stripe.go

```diff
@@ -351,17 +351,22 @@ func (s *BackendImplementation) Do(req *http.Request, body *bytes.Buffer, v inte
 			break
 		}
 
-		resBody, err := ioutil.ReadAll(res.Body)
-		res.Body.Close()
 		if err != nil {
 			if s.LogLevel > 0 {
-				s.Logger.Printf("Cannot read response: %v\n", err)
+				s.Logger.Printf("Request failed with error: %v\n", err)
+			}
+		} else {
+			resBody, err := ioutil.ReadAll(res.Body)
+			res.Body.Close()
+			if err != nil {
+				if s.LogLevel > 0 {
+					s.Logger.Printf("Cannot read response: %v\n", err)
+				}
+				return err
+			}
+			if s.LogLevel > 0 {
+				s.Logger.Printf("Request failed with: %s (error: %v)\n", string(resBody), err)
```

Comment:
@agatamigalska-newstore Sorry, I know this is code I gave you, but reading it again, I now realize that on this last log line `err` will never be non-`nil` because we're now handling that case above. It would also be kind of nice if we logged the response status code because that's one of the factors that the library will use to determine the retry.

Would you mind changing it to:

``` go
s.Logger.Printf("Request failed with body: %s (status: %v)\n", string(resBody), res.StatusCode)

Reply:
And thanks for making all those other changes!

After this last one, this looks good to me.

Reply:
Updated as requested, this time I also read it instead of just copy-pasting ;)